addons/sourcemod/scripting/nd_offmap_fixes.sp

```diff
@@ -31,54 +33,67 @@ int tmpAxisCount;
 int tmpAxisViolated;
 
 //Version is auto-filled by the travis builder
-public Plugin myinfo = 
+public Plugin myinfo =
 {
-	name 		= "[ND] Off Map Buildings Fixes",
-	author 		= "yed_, stickz",
-    	description 	= "Prevents building things in glitched locations",
-    	version 	= "dummy",
-	url 		= "https://github.com/stickz/Redstone/"
+    name        = "[ND] Off Map Buildings Fixes",
+    author      = "yed_, stickz, databomb",
+    description = "Prevents building things in glitched locations",
+    version     = "dummy",
+    url         = "https://github.com/stickz/Redstone/"
 }
 
 #define UPDATE_URL  "https://github.com/stickz/Redstone/raw/build/updater/nd_offmap_fixes/nd_offmap_fixes.txt"
 #include "updater/standard.sp"
 
-public void OnPluginStart() 
+// Check for nd_structure_intercept being available
+bool g_bStructureDetourAvailable = true;
+public void OnAllPluginsLoaded()
 {
-	HAX = new ArrayList(6);	
-    	AddUpdaterLibrary(); //auto-updater
+    g_bStructureDetourAvailable = LibraryExists("nd_structure_intercept");
+    if (!g_bStructureDetourAvailable)
+    {
+        LogError("Problem was found and automatically reverting to ND_OnStructureCreated. Check gamedata.");
```

Comment:
Needs clarity. How about: `Structure detour not available. Reverting to ND_OnStructureCreated. Check gamedata.`